source/org/zfin/marker/service/MarkerService.java

```diff
@@ -862,6 +862,26 @@ public static boolean markerHasAlias(Marker marker, String alias) {
         return false;
     }
 
+
+    public static MarkerAlias createMarkerAlias(Marker marker, String newAlias, List<String> publicationIDs) {
+        Iterator<String> pubIterator = publicationIDs.iterator();
+        Publication publication = publicationRepository.getPublication(pubIterator.next());
+
+        MarkerAlias alias = markerRepository.addMarkerAlias(marker, newAlias, publication);
+
+        // add direct attribution to aliased marker and provided publication
+        infrastructureRepository.insertStandardPubAttribution(marker.getZdbID(), publication);
+
+        while (pubIterator.hasNext()) {
```

Comment:
Why are you not including the first element's three method calls in the loop?

Reply:
If I understand correctly, you are asking why the first element is handled with:
```
        MarkerAlias alias = markerRepository.addMarkerAlias(marker, newAlias, publication);
        infrastructureRepository.insertStandardPubAttribution(marker.getZdbID(), publication);
```

while the rest of the elements are handled in the loop with this:
```
        markerRepository.addDataAliasAttribution(alias, publication, marker);
        infrastructureRepository.insertStandardPubAttribution(marker.getZdbID(), publication);
```

This logic was preserved from the previous version of the code.  The "addMarkerAlias" only needs to be called once and that method creates an alias attribution internally so we don't need to call "addDataAliasAttribution" on the first element.  I suppose it doesn't hurt to call "addDataAliasAttribution" twice since it has a check for duplicates.  I'll send another commit for consideration.